### PR TITLE
New Operator maturity model

### PR DESCRIPTION
**Pull request by dmesser**
**Description of the change:**

Updated graphics for Operator maturity model and new convention for `spec.maturity` in CSV to follow the model. Relevant documentation is updated.

**Motivation for the change:**

Provides a clearer understanding of the Operators capability rather than its stability level (which should be conveyed by semver).

### CODE REVIEW

File: doc/design/milestone-0.2.0/csv-generation.md [hunk 1]
```diff
@@ -157,7 +157,7 @@ Optional:
 - `spec.links`: 1..N URL's to websites, documentation, etc. pertaining to the operator or application being managed, each with a `name` and `url`.
 - `spec.selector`: selectors by which the operator can pair resources in a cluster.
 - `spec.icon`: a base64-encoded icon unique to the operator, set in a `base64data` field with a `mediatype`.
-- `spec.maturity`: the operators' stability, ex. `beta`.
+- `spec.maturity`: the operators capability level according to the [maturity model](https://github.com/operator-framework/operator-sdk/blob/master/doc/images/Operator-Maturity-Detailed.png), ex. `Seamless Upgrades`.
```
Comment (estroz) [inline doc/design/milestone-0.2.0/csv-generation.md]:
```suggestion
- `spec.maturity`: the operators' capability level according to the [operator maturity model](https://github.com/operator-framework/operator-sdk/blob/master/doc/images/Operator-Maturity-Detailed.png), ex. `Seamless Upgrades`.
```

File: pkg/scaffold/olm-catalog/csv.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ func (s *CSV) initCSVFields(csv *olmapiv1alpha1.ClusterServiceVersion) {
 	csv.Spec.Version = *semver.New(s.CSVVersion)
 	csv.Spec.DisplayName = getDisplayName(s.ProjectName)
 	csv.Spec.Description = "Placeholder description"
-	csv.Spec.Maturity = "alpha"
+	csv.Spec.Maturity = "Basic Install"
```
Comment (LiliC) [inline pkg/scaffold/olm-catalog/csv.go]: Is the term "Basic Install" documented anywhere or standardized, as to me it doesn't convey that much?

File: doc/images/Operator-Maturity-Detailed.svg [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (LiliC) [inline doc/images/Operator-Maturity-Detailed.svg]: Nit: can we change this to lower case file name instead? e.g. `operator-maturity-detailed.svg`

File: pkg/scaffold/olm-catalog/csv.go (hunk 2 above)
Comment (dmesser) [reply pkg/scaffold/olm-catalog/csv.go]: yes, we link to it in the csv-generation.md and in building-your-csv.md on OLM docs. It's basically describing the capability of an operator, i.e. what parts of an application lifecycle it's able to automate.

File: doc/images/Operator-Maturity-Detailed.svg (hunk 3 above)
Comment (dmesser) [reply doc/images/Operator-Maturity-Detailed.svg]: done

File: doc/design/milestone-0.2.0/csv-generation.md (hunk 1 above)
Comment (dmesser) [reply doc/design/milestone-0.2.0/csv-generation.md]: done

**Comment by LiliC** [comment]
Restarted the test job, seems to be another oocurance of this https://github.com/operator-framework/operator-sdk/issues/1103

**Comment by LiliC** [comment]
Besides the flaky test above, it's also the fact that it can't yet find the image you are refercing as it's not in master yet. 
```
Found broken url       (maturity model -> https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator-maturity-detailed.png) at doc/design/milestone-0.2.0/csv-generation.md:160 : 404 Not Found
```

**Comment by dmesser** [comment]
It's here https://github.com/operator-framework/operator-sdk/blob/master/doc/images/Operator-Maturity-Detailed.png

**Comment by LiliC** [comment]
@dmesser Yes, but the one you reference in the `csv-generation.md` is not the same one. :)

File: doc/images/Operator-Maturity-Detailed.svg (hunk 3 above)
Comment (LiliC) [reply doc/images/Operator-Maturity-Detailed.svg]: The filename is not renamed, also where is this file used?

**Comment by dmesser** [comment]
It was before. You proposed to change it to lower-case :)

**Comment by LiliC** [comment]
No, I suggested the newly added file be renamed to lowercase the `doc/images/Operator-Maturity-Detailed.svg`. :)

**Comment by AlexNPavel** [comment]
For files that are in the repo, we should be using relative links instead of URLs. So instead of `https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator-maturity-detailed.png`, we should use `../../images/operator-maturity-detailed.png`. This way we don't depend on a URL which may not be permanent (or in this case, may not exist yet)

File: doc/images/Operator-Maturity-Detailed.svg (hunk 3 above)
Comment (dmesser) [reply doc/images/Operator-Maturity-Detailed.svg]: sorry, file is now renamed - my local filesystem is case insensitive. The SVG is to preserve edit-ability.

**Comment by dmesser** [comment]
rebased

**Pull request merged**